File: daily_problem/2023-3/2395.cpp
/*
2395. 和相等的子数组
https://leetcode.cn/problems/find-subarrays-with-equal-sum/description/
*/

#include<vector>
#include<unordered_set>

using std::vector;
using std::unordered_set;

class Solution {
public:
    bool findSubarrays(vector<int>& nums) {
        unordered_set<int> s;
        for (int i = 0; i < nums.size() - 1; ++i) {
            if (s.count(nums[i] + nums[i + 1])) {
                return true;
            }
            s.insert(nums[i] + nums[i + 1]);
        }
        return false;
    }
};

File: daily_problem/2023-3/2395.py
"""
2395. 和相等的子数组
https://leetcode.cn/problems/find-subarrays-with-equal-sum/description/
"""

from itertools import pairwise
from typing import List


class Solution:
    def findSubarrays(self, nums: List[int]) -> bool:
        s = set()
        for x, y in pairwise(nums):
            if x + y in s:
                return True
            s.add(x + y)
        return False
